crates/matrix-sdk/src/room/edit.rs

```diff
@@ -167,47 +185,47 @@ async fn make_edit_event<S: EventSource>(
             };
 
             let mentions = original.content.mentions.clone();
+            let replied_to_original_room_msg =
+                extract_replied_to(source, room_id, original.content.relates_to).await;
+
+            let replacement = new_content.make_replacement(
+                ReplacementMetadata::new(event_id.to_owned(), mentions),
+                replied_to_original_room_msg.as_ref(),
+            );
+
+            Ok(replacement.into())
+        }
 
-            // Do a best effort at finding the replied-to original event.
-            let replied_to_sync_timeline_event =
-                if let Some(Relation::Reply { in_reply_to }) = original.content.relates_to {
-                    source
-                        .get_event(&in_reply_to.event_id)
-                        .await
-                        .map_err(|err| {
-                            warn!("couldn't fetch the replied-to event, when editing: {err}");
-                            err
-                        })
-                        .ok()
-                } else {
-                    None
-                };
-
-            let replied_to_original_room_msg = replied_to_sync_timeline_event
-                .and_then(|sync_timeline_event| {
-                    sync_timeline_event
-                        .raw()
-                        .deserialize()
-                        .map_err(|err| warn!("unable to deserialize replied-to event: {err}"))
-                        .ok()
-                })
-                .and_then(|event| {
-                    if let AnyTimelineEvent::MessageLike(AnyMessageLikeEvent::RoomMessage(
-                        MessageLikeEvent::Original(original),
-                    )) = event.into_full_event(room_id.to_owned())
-                    {
-                        Some(original)
-                    } else {
-                        None
-                    }
+        EditedContent::MediaCaption { caption, formatted_caption } => {
+            // Handle edits of m.room.message.
+            let AnySyncMessageLikeEvent::RoomMessage(SyncMessageLikeEvent::Original(original)) =
```

Comment:
Shouldn't we restrict this to actual media?

Reply:
That happens a few lines later under the function that edits the caption.

Reply:
Oh I see and then it would go on a `IncompatibleEditType`, fair enough 👍